### DISCUSSION TITLE
‘(not set)’,  ‘No value’ or ‘undefined’ properties

### DISCUSSION THREAD

**Discussion by enague** [mixpanel/docs-legacy#716] [Tips and Tricks]
## What does it mean?

Seeing properties marked as '(not set)', ‘undefined’ or ‘(No value)’ is expected behavior. All three of them mean the same: the property being queried or shown in the UI **was not ingested for that event or profile**; it is **not present** in that event/profile. 

Here’s an example: if you send an event to Mixpanel with a single property called ‘Brand’, when applying a breakdown by the property ‘Model’ you’ll see ‘(not set)’, as your event did not have any property called ‘Model’ in it.

Here’s how these not-present values look like in the Mixpanel UI:

![Untitled](https://github.com/mixpanel/docs/assets/35945133/a44833c3-01ab-4071-8b63-2e74203ec206)

![Untitled (1)](https://github.com/mixpanel/docs/assets/35945133/78efbf92-c0dc-45f3-b100-297846654c13)

## Why does it happen?

There are several reasons why a property can be missing on an event or profile. Here are the most common ones:

- **You aren’t tracking that property at that moment.** Imagine you have a signup form, but realize you forgot to ask people for their City upon signing up. You add a field for people to add their City, but you don’t have this data for previous respondents. This field will show as ‘(not set)’ in Mixpanel for those respondents signing up before your update.
- **The property was optional, so there’s no data to send to Mixpanel:** Following the example above: the field for City is there, but it’s optional. Some people completed that field, some didn’t. The empty fields will show as ‘(not set)’ for some of the users.

## How do I fix it?

If you were actually not expecting those “(not set)” values, it’s time to review your implementation of our SDK to make sure that in 100% of the cases the property is sent on the events/profiles you send our way.

Here’s a refresher, on how to set properties for Events and Users.

### Events

```jsx
// Send a "Played song" event to Mixpanel with a property "genre"
mixpanel.track("Played song",{"genre": "hip-hop"});
```

Here is the documentation for [sending Events with JavaScript](https://docs.mixpanel.com/docs/tracking-methods/sdks/javascript#sending-events). Choose your SDK on the left-hand sidebar.

### Users

To set User properties, Mixpanel uses `mixpanel.people.set()`:

```jsx
// mixpanel.identify must be called first to indicate which profile is being updated
mixpanel.identify("13793");

mixpanel.people.set({ "Plan": "Premium" });
```

This will set a "Plan" property, with a value "Premium", on user 13793's profile. If there isn't a profile with distinct_id 13793 in Mixpanel already, a new profile will be created. If that property was already set, the old value will be overwritten with "Premium".

[Here](https://docs.mixpanel.com/docs/tracking-methods/sdks/javascript#storing-user-profiles) is the documentation for setting Profile properties with JavaScript. Choose your SDK on the left-hand sidebar.

## How do I filter out unset values in a report?

There are 2 ways to discard from a report the events/properties without a certain property present:

### Method 1 - unselecting the “not set” segment

If using a breakdown: once you have queried your desired properties, unselect the ‘(not set)’ segment. Depending on the visualization the segments of a breakdown will appear under the chart or on the right side. Here’s an example [report](https://mixpanel.com/project/2144029/view/297551/app/insights#232N6RWDsu3D).

![Untitled (2)](https://github.com/mixpanel/docs/assets/35945133/1248b672-1130-4d08-b36d-3484056400e4)

### Method 2 - typecasting to True/False

1. In the Global Filter section of your report, choose your desired property
2. Click on the 3 dots > Data Type > True/False
3. The default option (True) will only show you values that are set

In other words “if [your property] is true” it means it’s present at the events/profiles being queried.

[This report](https://mixpanel.com/project/2144029/view/297551/app/insights#z4Npiwj5Y1t6) is an example of the end result. The gif below shows the above steps:

![Monosnap screencast 2023-11-17 11-02-35](https://github.com/mixpanel/docs/assets/35945133/e6c6af6e-e396-4b46-98b6-086172e938a4)
